core/src/main/scala/org/locationtech/rasterframes/RasterFunctions.scala

```diff
@@ -48,311 +48,328 @@ trait RasterFunctions {
+  def rf_array_to_tile(arrayCol: Column, cols: Int, rows: Int) = withAlias("rf_array_to_tile", arrayCol)(
     udf[Tile, AnyRef](F.arrayToTile(cols, rows)).apply(arrayCol)
   )
 
   /** Create a Tile from a column of cell data with location indexes and preform cell conversion. */
-  def assemble_tile(columnIndex: Column, rowIndex: Column, cellData: Column, tileCols: Int, tileRows: Int, ct: CellType): TypedColumn[Any, Tile] =
-    convert_cell_type(TileAssembler(columnIndex, rowIndex, cellData, lit(tileCols), lit(tileRows)), ct).as(cellData.columnName).as[Tile](singlebandTileEncoder)
+  def rf_assemble_tile(columnIndex: Column, rowIndex: Column, cellData: Column, tileCols: Int, tileRows: Int, ct: CellType): TypedColumn[Any, Tile] =
+    rf_convert_cell_type(TileAssembler(columnIndex, rowIndex, cellData, lit(tileCols), lit(tileRows)), ct).as(cellData.columnName).as[Tile](singlebandTileEncoder)
 
   /** Create a Tile from  a column of cell data with location indexes. */
-  def assemble_tile(columnIndex: Column, rowIndex: Column, cellData: Column, tileCols: Column, tileRows: Column): TypedColumn[Any, Tile] =
+  def rf_assemble_tile(columnIndex: Column, rowIndex: Column, cellData: Column, tileCols: Column, tileRows: Column): TypedColumn[Any, Tile] =
     TileAssembler(columnIndex, rowIndex, cellData, tileCols, tileRows)
 
   /** Extract the Tile's cell type */
-  def cell_type(col: Column): TypedColumn[Any, CellType] = GetCellType(col)
+  def rf_cell_type(col: Column): TypedColumn[Any, CellType] = GetCellType(col)
 
   /** Change the Tile's cell type */
-  def convert_cell_type(col: Column, cellType: CellType): TypedColumn[Any, Tile] =
+  def rf_convert_cell_type(col: Column, cellType: CellType): TypedColumn[Any, Tile] =
     SetCellType(col, cellType)
 
   /** Change the Tile's cell type */
-  def convert_cell_type(col: Column, cellTypeName: String): TypedColumn[Any, Tile] =
+  def rf_convert_cell_type(col: Column, cellTypeName: String): TypedColumn[Any, Tile] =
     SetCellType(col, cellTypeName)
 
+  /** Resample tile to different size based on scalar factor or tile whose dimension to match. Scalar less
+    * than one will downsample tile; greater than one will upsample. Uses nearest-neighbor. */
+  def rf_resample[T: Numeric](tileCol: Column, factorValue: T) = Resample(tileCol, factorValue)
+
+  /** Resample tile to different size based on scalar factor or tile whose dimension to match. Scalar less
+    * than one will downsample tile; greater than one will upsample. Uses nearest-neighbor. */
+  def rf_resample(tileCol: Column, factorCol: Column) = Resample(tileCol, factorCol)
+
   /** Convert a bounding box structure to a Geometry type. Intented to support multiple schemas. */
-  def extent_geometry(bounds: Column): TypedColumn[Any, Geometry] = ExtentToGeometry(bounds)
+  def rf_extent_geometry(bounds: Column): TypedColumn[Any, Geometry] = ExtentToGeometry(bounds)
```

Comment:
All extremely good comments. My reflections (please comment): 

* Thinking about it in terms of pairs of invertible functions is right on, and we should define both when appropriate.
* If the operations are to/from geometric types, perhaps we should stick with the `st_` prefix.
* If it is a single direction `Raster => Geometry` transformation, we use `rf_`.
* If the function takes the form `X_<preposition>...`, should we go with `X_<preposition>_Y` since Python and SQL don't have declared input types? That would argue _against_ **B**.
* Part of me leans toward just `<prefix>_X` for getter-type operations rather than involving a preposition.
* Selecting between `X_to_Y` and `Y_from_X` is tricky. In my mind `to` implies a conversion and `from` implies an extraction or getting operation.
* Is it a good thing to allow a function overloading across the geometry/raster domains, such that `rf_extent_from(Raster|Geometry)` is valid? Or should it be one of:
  - `rf_extent(Raster)` vs `st_extent(Geometry)`
  - `rf_extent_from(Raster)` vs `st_extent_from(Geometry)`
  - `rf_extent_from_raster(Raster)` vs `st_extent_from_geom(Geometry)`

All this is further complicated by the fact that there's a GeoMesa `st_envelope` function that gets the bounds of a `Geometry`, but returns it as another `Geometry`, and not the JTS `Envelope` type.

Reply:
Overall my preferences would be along the lines of:

* `rf_` for `Raster => T` 👍 
* don't use `from` preposition unless the function purpose is totally unclear otherwise
* preposition `to` for conversion 
* Since Python and SQL do not have declared input types, let's think about what happens if the user calls `rf_extent(Geometry)`. I think there would be a runtime error when an action is taken; or the expression evaluate to null. I think I'd like to see both prefixes offered but perhaps a single Expression under the hood that can handle several cases.

Marinating on all this I would propose:

`rf_extent(prt)`
`rf_geometry(prt)` 
`st_extent(geometry)`  dual with `st_geometry(extent)`

Reply:
* [ ] `rf_extent(prt)`
* [ ] `rf_geometry(prt)`
* [ ] `st_extent(geometry)`
* [ ] `st_geometry(extent)`

Reply:
adf229a5517a59b5248c4f39386253ab5e5ab759

Reply:
51c7686d7ec201faa7b75497db0b3404fa3c8954